### PR TITLE
Apache instrumentation v1.0

### PR DESCRIPTION
**Pull request by chrlic**
This PR proposes auto-instrumentation for applications running on Apache HTTPD server in K8S environment.

It also replaces earlier submitted PR https://github.com/open-telemetry/opentelemetry-operator/pull/1082

For instrumentation library, the logic provided here uses https://github.com/open-telemetry/opentelemetry-cpp-contrib/tree/main/instrumentation/otel-webserver-module
For instrumentation logic itself, it adheres to the principles used for other languages already supported as much as possible.

This PR is a first in a series as discussed in https://github.com/open-telemetry/opentelemetry-operator/pull/1082 and provides a method for building an image with webserver agent libraries. 

Next PR will provide CRD for Apache auto-instrumentation specification and related code.

And next PR will provide the code for auto-instrumentation logic within mutating webhook, related code, and end2end tests.

### CODE REVIEW

**Comment by chrlic** [comment]
@yuriolisa Since this PR is only about agent image build and does not do the injection itself, I do not see how to test it in the e2e test framework - as far as I can tell, all e2e tests rely currently on pre-build images. My plan was to include e2e test for the final PR. But, if you have a suggestion in this direction, I'm open to that.

File: autoinstrumentation/apache/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# How to build Apache auto-instrumentation docker image
```
Comment (pavolloffay) [inline autoinstrumentation/apache/README.md:1]: can we come up with a more explicit name (for the directory/image/readme name)? Apache is just too generic

**Review by pavolloffay — COMMENTED** [review]
The PR should include a github action to build and publish the image

**Comment by chrlic** [comment]
Renamed the "apache" to "apache-httpd"
Added github action for autoinstrumentation image build for apache-httpd

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**